Compile error when compiling with Lua 5.1

Hello,

I obtain a compile error on Ubuntu 15.10:

$ make
[  3%] Building CXX object CorsixTH/CMakeFiles/CorsixTH.dir/Src/rnc.cpp.o
In file included from /home/legluondunet/Applications_persos/Jeux/DOS/Theme_Hospital/Corsix-th/CorsixTH08122015/CorsixTH/Src/rnc.cpp:39:0:
/home/legluondunet/Applications_persos/Jeux/DOS/Theme_Hospital/Corsix-th/CorsixTH08122015/CorsixTH/Src/th_lua.h: In function ‘void luaT_register(lua_State*, const char*, const std::vector<luaL_Reg>&)’:
/home/legluondunet/Applications_persos/Jeux/DOS/Theme_Hospital/Corsix-th/CorsixTH08122015/CorsixTH/Src/th_lua.h:57:26: error: cannot convert ‘const std::vector<luaL_Reg>’ to ‘const luaL_Reg*’ for argument ‘3’ to ‘void luaL_register(lua_State*, const char*, const luaL_Reg*)’
     luaL_register(L, n, l);
                          ^
CorsixTH/CMakeFiles/CorsixTH.dir/build.make:77: recipe for target 'CorsixTH/CMakeFiles/CorsixTH.dir/Src/rnc.cpp.o' failed
make[2]: *** [CorsixTH/CMakeFiles/CorsixTH.dir/Src/rnc.cpp.o] Error 1
CMakeFiles/Makefile2:265: recipe for target 'CorsixTH/CMakeFiles/CorsixTH.dir/all' failed
make[1]: *** [CorsixTH/CMakeFiles/CorsixTH.dir/all] Error 2
Makefile:116: recipe for target 'all' failed
make: *** [all] Error 2

Thank you for your help.

LGDN

--- a/CorsixTH/Src/th_lua.h
+++ b/CorsixTH/Src/th_lua.h
@@ -54,7 +54,7 @@ inline void luaT_register(lua_State *L, const char *n, const std::vector<luaL_Re
     lua_pushvalue(L, -1);
     lua_setglobal(L, n);
 #else
-    luaL_register(L, n, l);
+    luaL_register(L, n, l.data());
 #endif
 }
 

